src/common/commonutils/UrlUtils.c

```diff
@@ -41,28 +43,23 @@ char* UrlEncode(char* target)
     return encodedTarget;
 }
 
-char* UrlDecode(char* target)
+char* UrlDecode(const char* target, size_t targetSize)
 {
-    int i = 0, j = 0;
+    size_t i = 0, j = 0;
     char buffer[3] = {0};
     unsigned int value = 0;
-    int targetLength = 0;
     char* decodedTarget = NULL;
 
-    if (NULL == target)
+    if ((NULL == target) || (0 == targetSize))
     {
         return NULL;
     }
 
-    targetLength = (int)strlen(target);
-    
-    // The length of the decoded string is the same as the length of the encoded string or smaller
-    decodedTarget = (char*)malloc(targetLength + 1);
-    if (NULL != decodedTarget)
+    if (NULL != (decodedTarget = (char*)malloc(targetSize + 3)))

```

Comment:
Why +3?

Reply:
To allow room for movement towards the end of the buffer. See loop on line 92. 

But in the end, see the other comment. These are not used for ASB. Only Encode is used and that's for a a very precise and legacy telemetry for IoT Hub. Any change there can regress that.